apps/meteor/app/file-upload/server/lib/FileUpload.js

```diff
@@ -115,7 +115,7 @@ export const FileUpload = {
 		try {
 			Promise.await(Apps.triggerEvent(AppEvents.IPreFileUpload, { file, content }));
 		} catch (error) {
-			if (error instanceof AppsEngineException) {
+			if (error.name === AppsEngineException.name) {
```

Comment:
Hi, @AllanPazRibeiro !

Why did you choice to use the string comparison instead of type checking? Looks weird

Reply:
Hey @dapi, this is because the actual code from the app runs inside an isolated context which requires every object it needs enclosed in a sandbox. This means that the error thrown by the app **will never be an instance of the `AppsEngineException` we have access to in our main context**. This was causing problems as Rocket.Chat wasn't able to determine what was an `AppsEngineException` anymore, and as such we decided to add a "type identifier" as a primitive value which we can assert